### ISSUE TITLE
owcc bug

### ISSUE THREAD

**Issue by stag019** [open-watcom/open-watcom-v2#144]
The POSIX driver (owcc) has (or had, I haven't tested with this version yet, just 1.9) a bug where it won't properly select the 16-bit compiler for 16-bit targets. More information on it is here: http://openwatcom.users.c-cpp.narkive.com/5Coeb0J6/owcc-posix-driver-bug-for-16-bit-applications

Has this been fixed/will it be fixed?

**Comment by ArmstrongJ** [comment]
I just tried owcc on GNU/Linux using a 7 March 2015 build, and it appears to be working:

```
jeff@centurion:/tmp$ owcc -v -bdos main.c
Open Watcom C/C++ x86 32-bit Compiler Driver Program
Version 2.0 beta Mar  7 2015 03:28:59 (32-bit)
Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1988-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See http://www.openwatcom.org/ for details.
	wcc -bt=dos -0 -fo=.o -fr main.c 
Open Watcom C x86 16-bit Optimizing Compiler
Version 2.0 beta Mar  7 2015 03:22:54 (32-bit)
Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1984-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See http://www.openwatcom.org/ for details.
main.c: 8 lines, included 748, 0 warnings, 0 errors
Code size: 21
	wlink @__owcc__.lnk 
Open Watcom Linker Version 2.0 beta Mar  7 2015 03:20:05 (32-bit)
Copyright (c) 2002-2015 The Open Watcom Contributors. All Rights Reserved.
Portions Copyright (c) 1985-2002 Sybase, Inc. All Rights Reserved.
Source code is available under the Sybase Open Watcom Public License.
See http://www.openwatcom.org/ for details.
loading object files
searching libraries
creating a DOS executable
jeff@centurion:/tmp$ 
```

**Comment by stag019** [comment]
Nice. Also, thanks for the quick reply.

Since you don't seem to mind using issues for random questions (#101), I'll ask this here too then. Do you have any sort of release cycle? I'd like to install the newest version, but I'd rather not have to take the time to compile it. How long does it take to compile everything, anyway? Is there any roadmap for what needs to be done before the unofficial official version 2.0 is complete (or beta or alpha or whathaveyou)?

**Comment by jmalak** [comment]
owcc in V2 fork was little reworked to properly handle selected architecture.
Jeff also enhance it to support Fortran compiler.

Now we are still in beta because not all parts of OW were ported to 64-bit hosts.
Main issue is C++ compiler, debug info browser and debugger.
As soon as these tools will be ported to 64-bit hosts I plan to build first release of this fork and pass it to production status. Next milestone is to modify code generator to be able generate 64-bit code for Intel architecture and modify front-ends for C and C++ compilers. I hope that first release will be available this year and next milestone 64-bit targets could be available next year.

**Comment by stag019** [comment]
Interesting...

I found the answer to one of my questions you didn't answer in BUILD.md ("A word of warning: running a full build may take upwards of two hours even on a fairly fast machine. There is a lot to build!").

Also, README.md mentions "Binary installers of this version of Open Watcom can be downloaded from SourceForge." However, the most recent version there is from 2014-04-28. Perhaps you could make a new version sooner than later, releasing it either there, and/or [here](https://github.com/open-watcom/open-watcom-v2/releases).

**Comment by ArmstrongJ** [comment]
I provide nightly builds via one of my company's [servers](http://buildbot.approximatrix.com/), but the builds are distributed only as zip files.  There are some additional environment variables that would need to be configured.  However, if you've installed Open Watcom once, you can basically drop the zip file contents over (or better yet, move the older build first) an existing install and everything will "just work" (except HLP files, I don't build those).  It's not ideal, but it suits my needs.

**Comment by stag019** [comment]
Sweet. That's exactly what I did with the 2014-04-28 build, but that hadn't had that fix yet. I've got the new build from today and that had the fix.

I think I'm going to close this and make a new issue about something else...
